### PR TITLE
feat: add `skipIf` option

### PR DESCRIPTION
**Pull request by jmcdo29**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

This was brought up on our [Discord](https://discord.com/channels/520622812742811698/819061892861526016/996086406986346536).

Currently there is (easy) way to provide an optional skip functionality when registering the `ThrottlerModule`. It would be possible to bind a `NoopGuard` in place of the `ThrottlerGuard`, but that does require writing extra code and deploying the unnecessary code in the end

## What is the new behavior?

There's a new `skipIf` option  that is a factory, taking in the `ExecutionContext` and returning a `boolean`

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: test/controller.e2e-spec.ts [hunk 1]
```diff
@@ -118,3 +119,37 @@ describe.each`
...
+          useClass: ThrottlerGuard,
+        },
+      ],
+    })
+      .overrideProvider(THROTTLER_OPTIONS)
+      .useValue({
+        skipIf: () => true,
+        limit: 5,
+      })
+      .compile();
```
Comment (Tony133) [inline test/controller.e2e-spec.ts:138]:
```suggestion
    .overrideProvider(THROTTLER_OPTIONS)
    .useValue({
      skipIf: () => true,
      limit: 5,
    })
    .compile();
```
```suggestion
      .overrideProvider(THROTTLER_OPTIONS)
      .useValue({
        skipIf: () => true,
        limit: 5,
      })
      .compile();
```

Comment (Tony133) [reply test/controller.e2e-spec.ts:138]: I think there is some indentation to fix here, but nothing problematic 😄

Comment (jmcdo29) [reply test/controller.e2e-spec.ts:138]: `yarn format` shows no changes need to be made

Comment (Tony133) [reply test/controller.e2e-spec.ts:138]: It will be github that has different indentation, however no problem 👍

**Pull request merged**